### DISCUSSION TITLE
Only files who match all the words

### DISCUSSION THREAD

**Discussion by blueray453** [BurntSushi/ripgrep#2441] [Q&A] [answered]
I have a command  which show files that match **any of the words**.

```
rg --files-with-matches --count-matches --sort path -i -e "use cases?" -e "user stor(y|ies)" -e "Technical debt" -e "Code Quality" -e "software development" -e "Agile Manifesto"
```

I want a command which will show files that match **all the words**.  All the patterns should have a match in the file (only then the file will the shown in the output).

Is there any solution for that?

**Comment by BurntSushi** [comment, marked as answer]
Probably what I would do is something like `rg foo -l | xargs rg bar -l | xargs rg quux -l`.

Otherwise, ripgrep doesn't currently support arbitrary boolean expressions. The typical way to find matching _lines_ containing `foo`, `bar` and `quux` is with a shell pipeline: `rg foo | rg bar | rg quux`.

See: https://github.com/BurntSushi/ripgrep/issues/875
